check/check.go

```diff
@@ -161,12 +162,19 @@ func (c *Check) Run() {
 		glog.V(2).Info(errmsgs)
 	}
 
-	res := c.Tests.Execute(out.String())
-	if res {
-		c.State = PASS
+	finalOutput := c.Tests.Execute(out.String())
+	if finalOutput != nil {
+		c.ActualValue = finalOutput.ActualResult
+
+		if finalOutput.TestResult {
+			c.State = PASS
+		} else {
+			c.State = FAIL
+		}
 	} else {
-		c.State = FAIL
+		return
```

Comment:
Hm, the user won't know that anything was wrong. c.State will be left as the zero value (I think - I didn't check thoroughly whether it is set elsewhere) so may even look like a pass. It's extremely unlikely to ever happen (lack of memory for allocating the struct) but if we're going to test for it, we should report on it - maybe exit with an error code too. 

Or write code so that it really can't happen :-)  

Also another little idiomatic piece of advice: if you are going to return (or exit) on error, it is generally considered clearer to handle the error case first, so you can have the "happy path" unindented. Something like: 

```
  if errorCondition {
     # report error
     return err      # or exit
  }

  # no else bracket needed - happy path goes here
```